gems/pending/disk/modules/miq_disk_cache.rb

```diff
@@ -0,0 +1,113 @@
+require 'rufus/lru'
+require_relative "../MiqDisk"
+require 'ostruct'
+
+module MiqDiskCache
+  MIN_SECTORS_TO_CACHE = 32
+  DEF_CACHE_SIZE       = 100
+  DEBUG_CACHE_STATS    = false
+
+  attr_reader :d_size, :blockSize, :cache_size, :sectors_to_cache, :cache_hits, :cache_misses
+
+  def self.new(down_stream, d_info = nil)
+    raise "MiqDiskCache: Downstream Disk Module is nil" if down_stream.nil?
+    @dInfo                  = d_info || OpenStruct.new
+    @dInfo.cache_size       = @dInfo.cache_size || DEF_CACHE_SIZE
```

Comment:
The name `cache_size` is vague and can be misleading. This is really the number of LRU entries.